### PR TITLE
Landing page file loading, Clickable columns on Categorization Page, and Vuex Store

### PR DESCRIPTION
**Pull request by jarmoza**
**Landing page and Categorization page now have initial functionality working**

1. Landing page allows for tsv and json file loading. At least the tsv file must be loaded to proceed to the categorization page.
2. Categorization page allows for the selection of categories in the listbox on the left and the subsequent "painting" of tsv columns in the table on the right. JSON description data is joined with the tsv column names in the table if available and semi-exact match is established between the category and column name.
3. Behind all of the above is a working Vuex store (index.js) which is communicated to from the pages (not components)

**Supporting components include:** 

a. the file-selector (landing page)
b. coloring-listgroup (listbox on categorization page), and 
c. the filedata-table (table on categorization page).

**Issues addressed:**

- Show the columns of a tsv file #2
- Make column headers clickable #3 
- Palette for painting of column headers and data dictionary fields #4 
- Have required categories selectable on the categorization screen #8 
- Store the classification of columns in the global store #18

### CODE REVIEW

**Comment by surchs** [comment]
awesome! Thanks @jarmoza for this PR!
I tried running the app and got it to load a .tsv and then I was able to paint the columns. Very nice!
The only odd thing is that when I am loading the .tsv, it seems to be only including the first column and discards all the rest. Any idea why that is / can you replicate that?

**Comment by jarmoza** [comment]
@surchs Can you attach here the tsv file you were using?

**Comment by surchs** [comment]
@jarmoza I used [this example file](https://github.com/metaneuro/annotation_tool/blob/master/assets/examples/good/ds003653_participant.tsv) but I think you can use any .tsv file with more than one column to reproduce.

I merged master into this branch to make these files available for testing for you.

**Comment by jarmoza** [comment]
@surchs Okay! I had to look in reverse in order to see what the issue was. There's nothing wrong with your file. There was something wrong with my test file(s) – which were apparently using spaces instead of tabs. The JS library I'm using to parse the file is apparently primarily built for csv files. So I'll have to figure out another solution for these tsv files. Stay tuned...

**Comment by surchs** [comment]
d3 seems to have things for this:
- https://github.com/d3/d3-dsv/tree/v3.0.1
- https://github.com/d3/d3/blob/main/API.md#delimiter-separated-values-d3-dsv

**Comment by jarmoza** [comment]
@surchs Thanks. I considered that. In fact, the current solution with papaparse produces correct data from the tsv data. There's actually just a processing error occurring in my file selector and store code. Fixing that now.

**Comment by surchs** [comment]
Alright, confirmed this works. That's great stuff @jarmoza! We're getting there!

File: pages/index.vue [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+			content-type="text/tab-separated-values"
+			v-on:file-selected="saveTsvFileData($event)">
+		</file-selector>
+
+		<!-- Debug component - shows file contents -->
 		<b-row>
-		    <textarea :rows="fileTextLines" cols="300">
-				{{ this.fileText.join("\n") }}
-			</textarea>
+			<h2>JSON File</h2>
```
Comment (surchs) [inline pages/index.vue:21]: Let's call this `Data dictionary` for now. Later we'll add some explanations.

**Review by surchs — APPROVED** [review]
Thanks @jarmoza, I think this looks very good. 

I also have a conceptual question:
- in your app, there are components and pages. Each of them combine HTML (in a <template> tag) and JS (in a <script> tag).
- This seems different from what I have seen in my (first and only) JS/Vue tutorial: https://github.com/Code-Pop/Intro-to-Vue-3/tree/final
  - there they have one big html page with the overall layout
  - and then there are several JS components, each with some html in there as well
and they all get imported from the big html file.

My question is: is there a reason you chose this layout? And do you think this has the right level of modularity for us to easily switch things around in the future / expand or even refocus? I don't have experience designing vue apps so I really don't have a good intuition on this

With this in mind I vote to merge this.

File: components/coloring-listgroup.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+				for ( let index = 0; index < listGroup.children.length; index++ ) {
+					
+					// A. Decolor the list group item
+					listGroup.children[index].style.backgroundColor = this.defaultPalette.bColor;
+					listGroup.children[index].style.color = this.defaultPalette.fColor;
+				}
+
+				// 4. Change the background and foreground colors of the clicked list group item
+								
+				// A. Color the clicked list group item
```
[the remark this answers is not in the document]
Comment (jarmoza) [reply components/coloring-listgroup.vue:70]: My bulleting system for commenting steps and substeps of a piece of code. 

0 - I use for setup
1-n Primary steps
A-Z Substeps for primary steps
I, II, III, etc. for Subsubsteps
a-z for subsubsubsteps 
i, ii, iii, etc. subsubsubsubsteps (if we ever really get that low)

File: pages/column-categorization.vue [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+				recommendedColumns: {
+					
+					names: [
+						"Subject ID",
+						"Age",
+						"Sex",
+						"Diagnosis",
+						"Assessment Tool"
+					],
+					backgroundColors: [
```
[the remark this answers is not in the document]
Comment (jarmoza) [reply pages/column-categorization.vue:68]: Yes, we could list that in the store.

File: pages/index.vue (hunk 1 above)
Comment (jarmoza) [reply pages/index.vue:21]: This change will be added in the a forthcoming PR.

**Pull request merged**